cdap-watchdog/src/main/java/io/cdap/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -523,13 +523,13 @@ private boolean writeFrequencyExceeded() {
     return false;
   }
 
-  private class PayloadInputStream extends ByteArrayInputStream {
+  public static class PayloadInputStream extends ByteArrayInputStream {
```

Comment:
This is hacky to turn a private inner class to public static. Better refactor this to a top level class in `cdap-common` and call it `ResettableByteArrayInputStream`.

Reply:
Done.